lib/api/core/httpRouter/index.js

```diff
@@ -158,48 +158,56 @@ class Router {
           .catch(error => replyWithError(cb, request, error));
       }
 
-      return replyWithError(cb, request, new BadRequestError(`Unrecognized HTTP method ${httpRequest.method}`));
+      return Promise.reject(replyWithError(cb, request, new BadRequestError(`Unrecognized HTTP method ${httpRequest.method}`)));
```

Comment:
what is the reason to reject the promise ?

Reply:
To ensure consistency on return value. This function always return a promise, either fulfilled or rejected but maybe it's not mandatory ?

Reply:
No, it doesn't. This method works with callbacks and not promises (see also comment from scottinet)